ARCHITECTURE.md

```diff
@@ -0,0 +1,108 @@
+
+Here's [a more complete guide to the engage pages and takeovers system](https://discourse.canonical.com/t/engage-pages-and-takeovers-v2/358).
+
+### Blog
+
+The blog pages under https://ubuntu.com/blog make use of [the blog module](https://github.com/canonical/canonicalwebteam.blog) ([instantiated here](webapp/app.py#L437-L442)) to pull in content from the `admin.insights.ubuntu.com` Wordpress installation through the Wordpress API. This content is managed through [the Wordpress admin area](https://admin.insights.ubuntu.com/admin), which is only accessible while on [the company VPN](https://wiki.canonical.com/InformationInfrastructure/IS/HowTo/CompanyOpenVPN).
+
+### Documentation
+
+There are a large number of documentation areas on ubuntu.com:
+
+- [ubuntu.com/servers/docs](http://ubuntu.com/servers/docs) ([code](webapp/app.py#L642-L651))
+- [ubuntu.com/core/services/guide](https://ubuntu.com/core/services/guide) ([code](webapp/app.py#L606-L616))
+- [ubuntu.com/core/docs](http://ubuntu.com/core/docs) ([code](webapp/app.py#L748-L756))
+  - Also [core sub docs sets](webapp/app.py#L757-L858): bluez, networkmanager, wpa-supplicant, easy-openvpn, wifi-ap, asla-utils
+- [ubuntu.com/openstack/docs](http://ubuntu.com/openstack/docs) ([code](webapp/app.py#L942-L952))
+- [ubuntu.com/security/livepatch/docs](http://ubuntu.com/security/livepatch/docs) ([code](webapp/app.py#L969-L979))
+- [ubuntu.com/security/certifications/docs](http://ubuntu.com/security/certifications/docs) ([code](webapp/app.py#L996-L1006))
+- [ubuntu.com/landscape/docs](http://ubuntu.com/landscape/docs) ([code](webapp/app.py#L1023-L1033))
+- [ubuntu.com/robotics/docs](http://ubuntu.com/robotics/docs) ([code](webapp/app.py#L1050-L1060))
+
+Each of these is served with [our Discourse module](https://github.com/canonical/canonicalwebteam.discourse), and pulls its content from a set of topics in Discourse, as with the takeovers and engage pages.
+
+For more information, see the [Creating Discourse based documentation pages](https://discourse.canonical.com/t/creating-discourse-based-documentation-pages/159) guide.
+
+### Search
+
+https://ubuntu.com/search ([url rule](webapp/app.py#L412-L421)) makes use of [our Google Programmable Search account](https://programmablesearchengine.google.com/u/1/controlpanel/overview?cx=009048213575199080868%3Ai3zoqdwqk8o) to pull results through Google's API using [our search module](https://github.com/canonical/canonicalwebteam.search/). The account is configured to index many different domains that we own, and results can be displayed from all of them.
+
+We have had trouble with search spam in the past which has led to us hitting API rate limits, breaking our search. There is now rate limiting on the content-cache level which will hopefully take care of this. We've set up [some dashboard in Graylog](https://logging.comms.canonical.com/dashboards/62cb3f41363defb179345f25) for monitoring the traffic on the search engine.
```

Comment:
```suggestion
We have had trouble with search spam in the past which has led to us hitting API rate limits, breaking our search. There is now rate limiting on the content-cache level which will hopefully take care of this. We've set up [some dashboards in Graylog](https://logging.comms.canonical.com/dashboards/62cb3f41363defb179345f25) for monitoring the traffic on the search engine.
```
Not sure if that's what you meant or if it's supposed to be "a dashboard" (also TIL I don't seem to have access to this 😅 )